Branch-SDK/Branch-SDK/BNCSystemObserver.m

```diff
@@ -121,13 +121,16 @@ + (NSString *)getDeviceName {
 }
 
 + (NSNumber *)getUpdateState {
-    NSString *bundleRoot = [[NSBundle mainBundle] bundlePath];    NSFileManager *manager = [NSFileManager defaultManager];
-    NSDictionary* attrs = [manager attributesOfItemAtPath:bundleRoot error:nil];
-    if ((int)([[attrs fileCreationDate] timeIntervalSince1970]/(60*60*24)) == (int)([[attrs fileModificationDate] timeIntervalSince1970]/(60*60*24))) {
-        return nil;
-    } else {
+
+    NSString *storedAppVersion = [BNCPreferenceHelper getAppVersion];
+    NSString *currentAppVersion = [BNCSystemObserver getAppVersion];
+
+    if (storedAppVersion == nil ||
+        ([storedAppVersion compare:currentAppVersion options:NSNumericSearch] == NSOrderedAscending)) {
+        [BNCPreferenceHelper setAppVersion:currentAppVersion];
```

Comment:
There wouldn't be anything saved because branch would actively track after
that fact, so that wouldn't count as an update, no?

I'd love to whiteboard the possible scenarios here tomorrow btw, think I'm
missing a few back end pieces here

On Wednesday, March 25, 2015, Alex Austin <notifications@github.com> wrote:

> In Branch-SDK/Branch-SDK/BNCSystemObserver.m
> <https://github.com/BranchMetrics/Branch-iOS-SDK/pull/66#discussion_r27189154>
> :
>
> > @@ -121,13 +121,16 @@ + (NSString *)getDeviceName {
> >  }
> >
> >  + (NSNumber *)getUpdateState {
> > -    NSString *bundleRoot = [[NSBundle mainBundle] bundlePath];    NSFileManager *manager = [NSFileManager defaultManager];
> > -    NSDictionary* attrs = [manager attributesOfItemAtPath:bundleRoot error:nil];
> > -    if ((int)([[attrs fileCreationDate] timeIntervalSince1970]/(60*60*24)) == (int)([[attrs fileModificationDate] timeIntervalSince1970]/(60*60*24))) {
> > -        return nil;
> > -    } else {
> > +
> > +    NSString *storedAppVersion = [BNCPreferenceHelper getAppVersion];
> > +    NSString *currentAppVersion = [BNCSystemObserver getAppVersion];
> > +
> > +    if (storedAppVersion == nil ||
> > +        ([storedAppVersion compare:currentAppVersion options:NSNumericSearch] == NSOrderedAscending)) {
> > +        [BNCPreferenceHelper setAppVersion:currentAppVersion];
>
> but what about the case when they first integrate Branch on a prexisting
> app with a large install base? We can't show all updates as installs
>
> The problem with today is that the app developer had 2k actual
> itunesconnect downloads and 200k upgrades. We counted 20k incorrectly as
> installs instead of updates
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/BranchMetrics/Branch-iOS-SDK/pull/66/files#r27189154>.
>